UI/Pages/Enquiry/Build/ConfirmSchool.cshtml.cs

```diff
@@ -0,0 +1,113 @@
+using Application.Common.Interfaces;
+using Application.Common.Models;
+using Application.Common.Models.Enquiry.Build;
+
+namespace UI.Pages.Enquiry.Build;
+
+public class ConfirmSchool : PageModel
+{
+    private readonly ISessionService _sessionService;
+    private readonly IMediator _mediator;
+
+    public ConfirmSchool(ISessionService sessionService, IMediator mediator)
+    {
+        _sessionService = sessionService;
+        _mediator = mediator;
+    }
+    [BindProperty] public ConfirmSchoolModel Data { get; set; } = new();
+
+    public bool FromSchoolPostcode { get; set; } = false;
+
+    public async Task<IActionResult> OnGetAsync(ConfirmSchoolModel data)
+    {
+        Data = data;
+
+        var isValid = await SetData(data);
+
+        if (!isValid)
+            return RedirectToPage(nameof(SchoolPostcode), data);
+
+        var schoolId = await _sessionService.GetAsync<int?>(SessionKeyConstants.EnquirySchoolId);
+
+        if (schoolId.HasValue &&
+            !data.Schools.Any(x => x.Id == schoolId.Value))
+            schoolId = null;
+
+        Data.SchoolId = schoolId;
+
```

Comment:
Adding `ModelState.Clear();` at this point is necessary. In our .cshtml file, we have implemented validation logic for the `asp-for="Data.SchoolId"` when multiple schools are displayed. Without clearing the ModelState, it will result in an error being shown on the page title.

Reply:
Good shout - didn't show the error on the UI, but just spotted that is is an error in the page.